Sources/SwiftGenKit/Parsers/Strings/StringsParser.swift

```diff
@@ -29,20 +29,25 @@ public enum Strings {
   }
 
   private enum Option {
-    static let separator = "separator"
+    static let separator = ParserOption(
+      key: "separator",
+      defaultValue: ".",
+      help: "Separator used to split keys into components"
+    )
   }
 
   public final class Parser: SwiftGenKit.Parser {
     var tables = [String: [Entry]]()
     public var warningHandler: Parser.MessageHandler?
     private let keyStructureSeparator: String
 
-    public init(options: [String: Any] = [:], warningHandler: Parser.MessageHandler? = nil) {
+    public init(options: [String: Any] = [:], warningHandler: Parser.MessageHandler? = nil) throws {
       self.warningHandler = warningHandler
-      self.keyStructureSeparator = (options[Option.separator] as? String) ?? Entry.defaultSeparator
+      self.keyStructureSeparator = try Option.separator.get(from: options)
```

Comment:
Thing is:
- Only 1 command has options for now (not to bad, we can add empty options to the other ones). Problem isn't now, but some future point where we have to remember that the init needs to parse/check the options.
- ParserOptionsList doesn't store values, it's a list of available options. We'd need an extra internal structure to store these.

Reply:
Ok, maybe `ParserOptionsList` wasn't the right type, but other than that, my whole point is if we implement such a think like in my comment, there's no way we'd risk avoiding checking options, because that check would have to be done if we want to extract those options from the dict in the first place


My point is, I really don't like the need of having that factory method, and having to remember to call it.

Tbh, I think there's way less risk that we forget to `check` the options during `init` — that could only happen if we implement a new parser one day, as I'd suggest we'd already add the call to `check` in the `init` of the existing parsers in that PR even the ones for which `allOptions = []`, after all we have to, to ensure they're not called with any option as they don't support any) — than the risk of not knowing / forgetting to call the `createWith` factory method (both for us in the various parts of SwiftGen & SwiftGenKit & SwiftGenKitTests where we have to create those parsers, than for any other app or framework that would want to use SwiftGenKit from their Swift code).

How many times do we create a new parser, for which there's that risk of forgetting to call `check`? And when we create new parsers we'll likely, if not copy/paste, at least get inspiration from existing Parsers. While for instantiating parsers, nobody would think that we shouldn't call `init` and will likely not read the doc-comment about it anyway, as long as autocomplete suggests a public init they'll call it without any 2nd thoughts and the risk of not using `createWith` like they'd be supposed to do is way more likely imho.